### PR TITLE
Issue #952 - implementing iterate() and get() methods in MkNotifications

### PR DESCRIPTION
**Pull request by prondzyn**
Related to issue #952. iterate() is based on empty list and get() method on simple MkNotification object.

### CODE REVIEW

**Comment by prondzyn** [comment]
@dmarkov please check the Travis build because the error does not look to be connected with my changes

**Comment by dmarkov** [comment]
@prondzyn Thanks, I will find someone to review this PR soon

**Comment by dmarkov** [comment]
@dmzaytsev it's yours, please review

File: src/main/java/com/jcabi/github/mock/MkNotifications.java [hunk 1]
```diff
@@ -40,24 +41,24 @@
...
  * @todo #913:30min Implement markAsRead() and thread() operations in
  *  MkNotifications. Don't forget about unit tests.
  */
 @Immutable
 final class MkNotifications implements Notifications {
+
     @Override
     public Iterable<Notification> iterate() {
-        throw new NotImplementedException("MkNotifications#iterate");
+        return Collections.emptyList();
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn I think you should return a real collection instead empty one.
Let's create the collection inside the constructor, size of the collection can be a parameter.

File: src/main/java/com/jcabi/github/mock/MkNotifications.java [hunk 2]
```diff
@@ -40,24 +41,24 @@
...
     }
 
     @Override
     public Notification get(final int number) {
-        throw new NotImplementedException("MkNotifications#get");
+        return new MkNotification(number);
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn you can return `number`th element of the collection

File: src/main/java/com/jcabi/github/mock/MkNotifications.java (hunk 1 above)
Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn you can fill the collection in constructor as well

File: src/test/java/com/jcabi/github/mock/MkNotificationsTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ *  iterate() operation in MkNotifications.
+ */
+public final class MkNotificationsTest {
+
+    /**
+     * MkNotifications can fetch empty list of Notification.
+     * @throws Exception if some problem inside
+     */
+    @Test
+    public void fetchEmptyListOfNotifications() throws Exception {
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/github/mock/MkNotificationsTest.java]: @prondzyn  `fetchesEmptyListOfNotifications`

File: src/test/java/com/jcabi/github/mock/MkNotificationsTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        MatcherAssert.assertThat(
+            new MkNotifications().iterate(),
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/github/mock/MkNotificationsTest.java]: @prondzyn `new MkNotifications(0).iterate()`

File: src/test/java/com/jcabi/github/mock/MkNotificationsTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            Matchers.emptyIterable()
+        );
+    }
+
+    /**
+     * MkNotifications can get single Notification.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void fetchSingleNotification() throws Exception {
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/github/mock/MkNotificationsTest.java]: @prondzyn why we need to check this case?
I think check for empty and non empty collection will be enough

File: src/test/java/com/jcabi/github/mock/MkNotificationsTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        Assert.assertNotNull(new MkNotifications().get(0));
+    }
+
+    /**
+     * MkNotifications can fetch non empty list of Notifications.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    @Ignore
+    public void fetchNonEmptyListOfNotifications() throws Exception  {
```
Comment (dmzaytsev) [inline src/test/java/com/jcabi/github/mock/MkNotificationsTest.java]: @prondzyn I guess it should a simple test, why you ignored it?

**Comment by dmzaytsev** [comment]
@prondzyn please see 7 comments/questions above

**Comment by prondzyn** [comment]
@dmzaytsev please check my improvements

File: src/main/java/com/jcabi/github/mock/MkNotifications.java [hunk 7]
```diff
@@ -40,24 +42,43 @@
...
+     * Public ctor.
+     * @param quantity Number of notifications.
+     * @checkstyle InnerAssignmentCheck (7 lines)
+     */
+    MkNotifications(final int quantity) {
+        int identifier = 0;
+        this.notifications = Collections
+            .<Notification>nCopies(
+                quantity,
+                new MkNotification(identifier += 1)
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn  I think you get `quantity` copies of `new MkNotification(1)`, right?
Let's initialize this one via `for` loop.

**Comment by dmzaytsev** [comment]
@prondzyn please see one comment above

File: src/main/java/com/jcabi/github/mock/MkNotifications.java (hunk 7 above)
Comment (prondzyn) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @dmzaytsev sorry, my mistake. 

But when I use `for` I need to add `@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")`. Is it not a problem?

Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn sure, you can suppress this warning

**Comment by prondzyn** [comment]
@dmzaytsev please check my fix now

File: src/main/java/com/jcabi/github/mock/MkNotifications.java [hunk 8]
```diff
@@ -40,24 +42,41 @@
...
+    private final transient List<Notification> notifications;
+
+    /**
+     * Public ctor.
+     * @param quantity Number of notifications.
+     */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    MkNotifications(final int quantity) {
+        this.notifications = new ArrayList<Notification>(quantity);
+        for (int index = 0; index < quantity; index += 1) {
```
Comment (dmzaytsev) [inline src/main/java/com/jcabi/github/mock/MkNotifications.java]: @prondzyn `++ index`

**Comment by dmzaytsev** [comment]
@prondzyn looks good just one minor remark

**Comment by prondzyn** [comment]
@dmzaytsev updated as you requested. Thanks for all your remarks. I still need to get familiar with your coding style.

**Comment by dmzaytsev** [comment]
@prondzyn thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge again

**Comment by dmzaytsev** [comment]
@yegor256 there aren't any test failures...
I don't understand why Rultor can't merge

**Comment by yegor256** [comment]
@dmzaytsev there are test failures, they are just in the middle of the build script

**Comment by dmzaytsev** [comment]
@yegor256 do you mean this one?

>[0;33mWARNm] pool-1-thread-16 org.reflections.Reflections: could not create Vfs.Dir from url. ignoring the exception and continuing org.reflections.ReflectionsException: could not create Vfs.Dir from url, no matching UrlType was found [file:/usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext/libatk-wrapper.so] either use fromURL(final URL url, final List<UrlType> urlTypes) or use the static setDefaultURLTypes(final List<UrlType> urlTypes) or addDefaultURLTypes(UrlType urlType) with your specialized UrlType. at org.reflections.vfs.Vfs.fromURL(Vfs.java:108)

how @prondzyn can fix it?

**Comment by darkled** [comment]
@dmzaytsev I think the problem is here:
```
[0;31mERRORm] pool-137-thread-26 com.jcabi.github.User$Smart: #name(): thrown java.lang.ClassCastException(javax.json.JsonValue$1 cannot be cast to javax.json.JsonString) out of org.glassfish.json.JsonObjectBuilderImpl$JsonObjectImpl#getJsonString[194] in 55ms Tests run: 8, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.164 sec <<< FAILURE! - in com.jcabi.github.RtIssueITCase
```

**Comment by dmzaytsev** [comment]
@darkled thank you

@prondzyn we have broken integration test `com.jcabi.github.RtIssueITCase`

[0;31mERRORm] pool-137-thread-26 com.jcabi.github.User$Smart: #name(): thrown java.lang.ClassCastException(javax.json.JsonValue$1 cannot be cast to javax.json.JsonString) out of org.glassfish.json.JsonObjectBuilderImpl$JsonObjectImpl#getJsonString[194] in 55ms
Tests run: 8, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.164 sec <<< FAILURE! - in com.jcabi.github.RtIssueITCase

**Comment by prondzyn** [comment]
@dmzaytsev I found an issue (#1178) in code which breaks the integration test. It is not connected which changes I committed.

@dmarkov This PR is blocked by issue I found and reported - #1178

**Comment by dmzaytsev** [comment]
@yegor256 @prondzyn  I propose ignore `com.jcabi.github.RtIssueITCase` and merge this PR

@prondzyn if we will do it please mention in  #1178 about the ignored test

**Comment by prondzyn** [comment]
@dmzaytsev I ignored 1 test from RtIssueITCase  and updated description of #1178

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by yegor256** [comment]
@prondzyn if you ignore any test, don't forget to add some description there, explaining why the test is ignored. Everybody should be able to understand why `@Ignore` is there, in a year, for example.

**Comment by prondzyn** [comment]
@yegor256 Understood. Comment added.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by prondzyn** [comment]
@yegor256 what should I do now? It is not in part which I did not touch.

**Comment by yegor256** [comment]
@prondzyn just fix it and move on. It should be very easy to fix, see https://github.com/teamed/pdd

**Comment by prondzyn** [comment]
@yegor256 should work now

**Comment by prondzyn** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by prondzyn** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by dmarkov** [comment]
@dmzaytsev Thanks a lot, I just topped your account for **26 mins**, transaction ID `67397152`, total time was 288 hours and 46 mins. extra minutes for review comments (c=11). +26 to your rating, your total score is [+1343](http://www.netbout.com/b/35166?open=rating)

**Comment by dmarkov** [comment]
@rultor please deploy

**Pull request merged**